### ISSUE TITLE
Plugin "itu-radio-material" not found!

### ISSUE THREAD

**Issue by llyaljl** [NVlabs/sionna#1136]
Hello, why can't I find the itu-radio-material when loading the .xml scene exported with Blender?

```py
# 步骤 1：检查 Sionna 版本
import sionna.rt
print(f"sionna.rt 版本：{sionna.rt.__version__}")

# 步骤 2：检查是否支持可微分 RT
import mitsuba as mi
from sionna.rt import load_scene

# 先测试标准变体
mi.set_variant('cuda_ad_rgb')  # 非可微分
try:
    scene = load_scene("room/room.xml")
    print("✅ 标准变体加载成功")
except Exception as e:
    print(f"❌ 标准变体失败：{e}")

# 再测试可微分变体
mi.set_variant('cuda_ad_rgb')
try:
    scene_ad = load_scene("room/room.xml")
    print("✅ 可微分变体加载成功")
except Exception as e:
    print(f"❌ 可微分变体失败：{e}")
```

output：sionna.rt 版本：1.2.1
❌ 标准变体失败：[parser.cpp:1708] At string (line 2, col 6): failed to instantiate bsdf plugin of type "itu-radio-material": [PluginManager] Plugin "itu-radio-material" not found!
❌ 可微分变体失败：[parser.cpp:1708] At string (line 2, col 6): failed to instantiate bsdf plugin of type "itu-radio-material": [PluginManager] Plugin "itu-radio-material" not found!

This .xml file：

```xml
<scene version="2.1.0">
    <bsdf type="sionna_radio_material" id="itu_brick">
        <string name="material_name" value="brick"/>
        <float name="relative_permittivity" value="4.44"/>
        <float name="electrical_conductivity" value="0.001"/>
    </bsdf>

    <bsdf type="sionna_radio_material" id="itu_wood">
        <string name="material_name" value="wood"/>
        <float name="relative_permittivity" value="3.0"/>
        <float name="electrical_conductivity" value="0.01"/>
    </bsdf>

    <bsdf type="sionna_radio_material" id="itu_chipboard">
        <string name="material_name" value="chipboard"/>
        <float name="relative_permittivity" value="2.9"/>
        <float name="electrical_conductivity" value="0.02"/>
    </bsdf>

    <bsdf type="sionna_radio_material" id="itu_concrete">
        <string name="material_name" value="concrete"/>
        <float name="relative_permittivity" value="5.31"/>
        <float name="electrical_conductivity" value="0.0326"/>
    </bsdf>

    <bsdf type="sionna_radio_material" id="itu_marble">
        <string name="material_name" value="marble"/>
        <float name="relative_permittivity" value="7.0"/>
        <float name="electrical_conductivity" value="0.002"/>
    </bsdf>

    <shape type="ply" id="floor">
        <string name="filename" value="meshes/floor.ply"/>
        <ref id="itu_brick"/>
    </shape>
    <shape type="ply" id="walls">
        <string name="filename" value="meshes/walls.ply"/>
        <ref id="itu_brick"/>
    </shape>
    <shape type="ply" id="walls_001">
        <string name="filename" value="meshes/walls_001.ply"/>
        <ref id="itu_brick"/>
    </shape>
    <shape type="ply" id="walls_002">
        <string name="filename" value="meshes/walls_002.ply"/>
        <ref id="itu_brick"/>
    </shape>
    <shape type="ply" id="walls_003">
        <string name="filename" value="meshes/walls_003.ply"/>
        <ref id="itu_brick"/>
    </shape>

    <shape type="ply" id="box1">
        <string name="filename" value="meshes/立方体.ply"/>
        <ref id="itu_wood"/>
    </shape>
    <shape type="ply" id="box2">
        <string name="filename" value="meshes/立方体_001.ply"/>
        <ref id="itu_chipboard"/>
    </shape>
    <shape type="ply" id="box3">
        <string name="filename" value="meshes/立方体_002.ply"/>
        <ref id="itu_concrete"/>
    </shape>
    <shape type="ply" id="box4">
        <string name="filename" value="meshes/立方体_003.ply"/>
        <ref id="itu_marble"/>
    </shape>

</scene>
```

**Comment by merlinND** [comment]
Hello @llyaljl,

The XML that you shared specifies `type="sionna_radio_material"`, but the error message mentions `itu-radio-material`. Are you sure that you are trying to load the same XML that you shared?

For reference, you can always check the built-in Sionna RT scenes and check for differences to your scene: https://github.com/NVlabs/sionna-rt/blob/39e25ebab8d7eaf468046748c79dbdfc8015bb3f/src/sionna/rt/scenes/box_two_screens/box_two_screens.xml

You can also check to make sure that the built-in scenes load correctly.

**Comment by tiamilani** [comment]
I have the same problem but with a minimal example when trying to load the built-in munich scene.

Below the minimal example:
```
import mitsuba as mi
import torch
import sionna

# Import relevant components from Sionna RT
from sionna.rt import load_scene

if torch.cuda.is_available():
    print("CUDA GPU detected.")
else:
    print("No CUDA GPU detected")

mi.set_variant("cuda_ad_rgb")

scene = load_scene(sionna.rt.scene.munich)
```

I'm running the code in a virtual environment with the following libraries and version:
```
asttokens==3.0.1
comm==0.2.3
contourpy==1.3.3
cuda-bindings==13.2.0
cuda-pathfinder==1.5.0
cuda-toolkit==13.0.2
cycler==0.12.1
decorator==5.2.1
drjit==1.3.1
executing==2.2.1
filelock==3.25.2
fonttools==4.62.1
fsspec==2026.2.0
h5py==3.16.0
importlib_resources==6.5.2
ipydatawidgets==4.3.5
ipython==9.11.0
ipython_pygments_lexers==1.1.1
ipywidgets==8.1.8
jedi==0.19.2
Jinja2==3.1.6
jupyterlab_widgets==3.0.16
kiwisolver==1.5.0
MarkupSafe==3.0.3
matplotlib==3.10.8
matplotlib-inline==0.2.1
mitsuba==3.8.0
mpmath==1.3.0
networkx==3.6.1
numpy==2.4.3
nvidia-cublas==13.1.0.3
nvidia-cuda-cupti==13.0.85
nvidia-cuda-nvrtc==13.0.88
nvidia-cuda-runtime==13.0.96
nvidia-cudnn-cu13==9.19.0.56
nvidia-cufft==12.0.0.61
nvidia-cufile==1.15.1.6
nvidia-curand==10.4.0.35
nvidia-cusolver==12.0.4.66
nvidia-cusparse==12.6.3.3
nvidia-cusparselt-cu13==0.8.0
nvidia-nccl-cu13==2.28.9
nvidia-nvjitlink==13.0.88
nvidia-nvshmem-cu13==3.4.5
nvidia-nvtx==13.0.85
packaging==26.0
parso==0.8.6
pexpect==4.9.0
pillow==12.1.1
prompt_toolkit==3.0.52
ptyprocess==0.7.0
pure_eval==0.2.3
Pygments==2.19.2
pyparsing==3.3.2
python-dateutil==2.9.0.post0
pythreejs==2.4.2
scipy==1.17.1
setuptools==81.0.0
sionna==2.0.0
sionna-rt==1.2.2
six==1.17.0
stack-data==0.6.3
sympy==1.14.0
torch==2.11.0
traitlets==5.14.3
traittypes==0.2.3
triton==3.6.0
typing_extensions==4.15.0
wcwidth==0.6.0
widgetsnbextension==4.0.15
```

The system is a DGX Spark with the following output from `nvidia-smi`:
```
Wed Mar 25 10:23:47 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 590.48.01              Driver Version: 590.48.01      CUDA Version: 13.1     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GB10                    On  |   0000000F:01:00.0 Off |                  N/A |
| N/A   40C    P8              5W /  N/A  | Not Supported          |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            2545      G   /usr/lib/xorg/Xorg                       19MiB |
|    0   N/A  N/A            2604      G   /usr/bin/gnome-shell                      6MiB |
+-----------------------------------------------------------------------------------------+
```

Please also notice that if the line `mi.set_variant("cuda_ad_rgb")` is commented out then the code executes without problems.

Could this be a mitsuba problem rather than sionna?

**Comment by merlinND** [comment]
I had not noticed the line `mi.set_variant('cuda_ad_rgb')` in @llyaljl's original code. **It should be removed**. Sionna already sets the correct Mitsuba variant upon import.

@llyaljl & @tiamilani, why did you add `mi.set_variant('cuda_ad_rgb')`? It is not part of any of our tutorial notebooks or documentation and should not be used.

**Comment by tiamilani** [comment]
Hi @merlinND thanks for the quick reply.

it is mentioned among the key topics in mitsuba documentation: [https://mitsuba.readthedocs.io/en/stable/src/key_topics/variants.html](https://mitsuba.readthedocs.io/en/stable/src/key_topics/variants.html).

Speaking for myself, I wanted to be sure that the system was configured to work with the GPU. That's why I first tested with torch that a GPU was correctly detected and later I wanted to instruct mitsuba to use it.

How can I check the version selected from sionna?

**Comment by tiamilani** [comment]
Found using `mitsuba.variant()` by default on my system `cuda_ad_mono_polarized` has been set.

**Comment by merlinND** [comment]
I see, thanks for the explanations @tiamilani!
Letting Sionna set the Mitsuba variant, and then checking with `mitsuba.variant()` if needed is the correct way :+1:

**Comment by llyaljl** [comment]
> I had not noticed the line in [@llyaljl](https://github.com/llyaljl)'s original code. **It should be removed**. Sionna already sets the correct Mitsuba variant upon import.`mi.set_variant('cuda_ad_rgb')`
> 
> [@llyaljl](https://github.com/llyaljl) & [@tiamilani](https://github.com/tiamilani), why did you add ? It is not part of any of our tutorial notebooks or documentation and should not be used.`mi.set_variant('cuda_ad_rgb')`

iI am very thanks for the explanations @tiamilani!

**Comment by llyaljl** [comment]
> 我之前没注意到原始代码里有这句话。**应该被移除**。Sionna在导入时已经设置了正确的Mitsuba变体。`mi.set_variant('cuda_ad_rgb')`
> 
> 以及，你为什么要加？它不属于我们任何教程笔记本或文档，也不应使用。`mi.set_variant('cuda_ad_rgb')`

I have a new question: Sionna RT is built on Mitsuba 3 and Dr.Jit. These frameworks provide automatic differentiation and can backpropagate gradients throughout the entire system. I want to use the built-in automatic differentiation of Sionna RT. How should I use it? Can you provide a reference?

**Comment by tiamilani** [comment]
Hi @llyaljl sorry but I personally don't have an answer for you, but I would suggest you to post your questions in the discussions section of the repository. There more knowledgeable people might be capable to easily guide you ;)
